src/sparsezoo/refactor/model_directory.py

```diff
@@ -0,0 +1,516 @@
+        `src.sparsezoo.requests.download.download_model_get_request()` function.
+
+    - `from_directory()` -> by using the directory to the model on your machine.
+
+    :param files: list of files, where every file
+        is represented by a dictionary (note: not Fil object)
+    :directory_path: if ModelDirectory created using method `from_directory()`,
+        it points to the directory of the ModelDirectory. By default: None
+    """
+
+    ATTRIBUTES = [
+        "framework_files",
+        "sample_originals",
+        "sample_inputs",
+        "sample_outputs",
+        "sample_labels",
+        "onnx_model",
+        "onnx_models",
+        "analysis",
+        "benchmarks",
+        "eval_results",
+        "model_card",
+        "recipes",
+    ]
+
+    def __init__(
+        self, files: List[Dict[str, Any]], directory_path: Optional[str] = None
+    ):
+
+        self.framework_files: FrameworkFiles = self._directory_from_files(
+            files,
+            directory_class=FrameworkFiles,
+            display_name="framework-files",
+        )
+        self.sample_originals: SampleOriginals = self._directory_from_files(
+            files,
+            directory_class=SampleOriginals,
+            display_name="sample-originals",
+        )
+        self.sample_inputs: NumpyDirectory = self._directory_from_files(
+            files,
+            directory_class=NumpyDirectory,
+            display_name="sample-inputs",
+        )
+        # TODO: Ignoring the dictionary type for now.
+        self.sample_outputs: Dict[str, NumpyDirectory] = self._directory_from_files(
+            files,
+            directory_class=NumpyDirectory,
+            display_name="sample-outputs",
+        )  # key by engine name.
+        self.sample_labels: Directory = self._directory_from_files(
+            files, directory_class=Directory, display_name="sample-labels"
+        )
+        self.onnx_model: File = self._file_from_files(
+            files, display_name="model.onnx"
+        )  # model.onnx
+
+        self.onnx_models: List[File] = self._file_from_files(
+            files, display_name="model.(.*).onnx", regex=True
+        )  # model{.opset}.onnx
+
+        self.analysis: File = self._file_from_files(
+            files, display_name="analysis.yaml"
+        )  # analysis.yaml
+        self.benchmarks: File = self._file_from_files(
+            files, display_name="benchmarks.yaml"
+        )  # benchmarks.yaml
+        self.eval_results: File = self._file_from_files(
+            files, display_name="eval.yaml"
+        )  # eval.yaml
+        self.model_card: File = self._file_from_files(
+            files, display_name="model.md"
+        )  # model.md
+        self.recipes: List[File] = self._file_from_files(
+            files, display_name="recipe(.*).md", regex=True
+        )  # recipe{_tag}.md
+
+        self._directory_path = directory_path
+
+    def __iter__(self) -> Tuple[str, Union[File, List[File], Dict]]:
```

Comment:
what would be the use case for this iter func?

Reply:
See the answer below.